### PR TITLE
Create a data product of cost per setting per year

### PR DESCRIPTION
**Pull request by otfrom**
directly via the histograms rather than made up from summing things from an output file.

### CODE REVIEW

File: src/clj/witan/send/model/data_products.clj [hunk 1]
```diff
@@ -74,6 +74,24 @@
    {}
    calendar-year))
 
+(defn roll-up-calendar-year-by-cost-per-setting [cost-lookup calendar-year]
+  (transduce
+   (comp
+    (remove (fn [[[ay need-setting] population]]
+              (= need-setting constants/non-send))) ;; remove non-send
+    ;; convert need-setting population to need-setting cost
+    (map (fn [[[ay need-setting] population]] [need-setting population]))
```
Comment (seb231) [inline src/clj/witan/send/model/data_products.clj]: So is the product of each of these 'map' functions just the data reformatted?

File: src/clj/witan/send/model/data_products.clj [hunk 2]
```diff
@@ -74,6 +74,24 @@
...
+    (map (fn [[need-setting population]] [(states/split-need-setting need-setting) population]))
+    (map (fn [[[need setting :as need-setting] population]] [setting
+                                                             (* (get cost-lookup need-setting 0)
+                                                                population)]))
```
Comment (seb231) [inline src/clj/witan/send/model/data_products.clj]: this looks simple but good 👍

File: doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Create new data slices at the simulation levels
+
+## Context
+
+The model projects future activity by creating multiple simulations of
+the future. We then run these simulation runs through a HDR Histogram
```
Comment (seb231) [inline doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md]: change to "run these simulations" or "run these simulation results"

File: doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+to get summary statistics. When we use the data directly at the
+ay/need/setting level, then we don't need to roll it up.
```
Comment (seb231) [inline doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md]: we don't need to roll it up to do what?

File: doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+sound, but means that we have to do a larger chunk of work for each
+new data product.
+
+### New data products need to be built off the simulation data rather than the summaries
+
+Because of how the HDR Histogram and the simulations work together, it
+isn't mathematically sound to create new data products by using any
+summary results. If you had the median/mean/iqr of the academic
+years/need/settings of the simulation, you couldn't roll that up to
+totals for need/setting without the academic years.
```
Comment (seb231) [inline doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md]: 👍

File: doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md (hunk 3 above)
Comment (otfrom) [reply doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md]: fixed

File: doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md (hunk 4 above)
Comment (otfrom) [reply doc/dsdr/0001-creating-new-slices-at-the-simulation-level.md]: fixed

File: src/clj/witan/send/model/data_products.clj [hunk 6]
```diff
@@ -187,29 +156,28 @@
   (let [by-state (future (summarise-results
                           (partial roll-up-calendar-year-by-state
                                    (states/calculate-valid-states-from-setting-academic-years valid-states))
-                          simulations))
+                          simulations)) ;; FIXME - this should be a state filter for all data products
```
Comment (seb231) [inline src/clj/witan/send/model/data_products.clj]: What does this mean?

File: src/clj/witan/send/model/data_products.clj [hunk 7]
```diff
@@ -187,29 +156,28 @@
...
         total-in-send-by-ay (future (summarise-results roll-up-calendar-year-by-ay simulations))
-        total-in-send (future (into []
-                                    (map :total-population)
-                                    (summarise-results roll-up-calendar-year-by-total-send-population simulations)))
-        total-in-send-by-need (future (summarise-results roll-up-calendar-year-by-total-in-send-by-need simulations))
-        total-in-send-by-setting (future (summarise-results roll-up-calendar-year-by-total-in-send-by-setting simulations))
+        total-in-send-by-ay-group (future (summarise-results roll-up-total-in-send-by-ay-group simulations))
```
Comment (seb231) [inline src/clj/witan/send/model/data_products.clj]: Any reason this was moved?

Comment (otfrom) [reply src/clj/witan/send/model/data_products.clj]: just to group things together by what I think will be the namespaces

File: src/clj/witan/send/model/data_products.clj (hunk 6 above)
Comment (otfrom) [reply src/clj/witan/send/model/data_products.clj]: I think we probably want this filter everywhere

Comment (seb231) [reply src/clj/witan/send/model/data_products.clj]: sorry still don't follow?

**Pull request merged**